packages/dtslint/src/checks.ts

```diff
@@ -69,76 +29,69 @@ export interface DefinitelyTypedInfo {
-      const expected = mustHave[key];
-      const actual = options[key];
-      if (!deepEquals(expected, actual)) {
-        throw new Error(
-          `Expected compilerOptions[${JSON.stringify(key)}] === ${JSON.stringify(expected)}, but got ${JSON.stringify(
-            actual
-          )}`
-        );
-      }
+export function checkTsconfig(dirPath: string, options: CompilerOptionsRaw): string[] {
+  const errors = [];
+  const mustHave = {
+    noEmit: true,
+    forceConsistentCasingInFileNames: true,
+    types: [],
+  };
+
+  for (const key of Object.getOwnPropertyNames(mustHave) as (keyof typeof mustHave)[]) {
+    const expected = mustHave[key];
+    const actual = options[key];
+    if (!deepEquals(expected, actual)) {
+      errors.push(
+        `Expected compilerOptions[${JSON.stringify(key)}] === ${JSON.stringify(expected)}, but got ${JSON.stringify(
+          actual
+        )}`
+      );
     }
+  }
 
-    for (const key in options) {
-      switch (key) {
-        case "lib":
-        case "noImplicitAny":
-        case "noImplicitThis":
-        case "strict":
-        case "strictNullChecks":
-        case "noUncheckedIndexedAccess":
-        case "strictFunctionTypes":
-        case "esModuleInterop":
-        case "allowSyntheticDefaultImports":
-        case "paths":
-        case "target":
-        case "jsx":
-        case "jsxFactory":
-        case "experimentalDecorators":
-        case "noUnusedLocals":
-        case "noUnusedParameters":
-        case "exactOptionalPropertyTypes":
-        case "module":
-          break;
-        default:
-          if (!(key in mustHave)) {
-            throw new Error(`Unexpected compiler option ${key}`);
-          }
-      }
+  for (const key in options) {
+    switch (key) {
+      case "lib":
+      case "noImplicitAny":
+      case "noImplicitThis":
+      case "strict":
+      case "strictNullChecks":
+      case "noUncheckedIndexedAccess":
+      case "strictFunctionTypes":
+      case "esModuleInterop":
+      case "allowSyntheticDefaultImports":
+      case "target":
+      case "jsx":
+      case "jsxFactory":
+      case "experimentalDecorators":
+      case "noUnusedLocals":
+      case "noUnusedParameters":
+      case "exactOptionalPropertyTypes":
+      case "module":
+      case "paths":
+        break;
+      default:
+        if (!(key in mustHave)) {
+          errors.push(`Unexpected compiler option ${key}`);
+        }
     }
   }
-
   if (!("lib" in options)) {
-    throw new Error('Must specify "lib", usually to `"lib": ["es6"]` or `"lib": ["es6", "dom"]`.');
+    errors.push('Must specify "lib", usually to `"lib": ["es6"]` or `"lib": ["es6", "dom"]`.');
   }
 
   if (!("module" in options)) {
-    throw new Error('Must specify "module" to `"module": "commonjs"` or `"module": "node16"`.');
-  }
-  if (
+    errors.push('Must specify "module" to `"module": "commonjs"` or `"module": "node16"`.');
```

Comment:
Huh, since when could we do `node16` resolution?

Reply:
uhh, I just remember porting this from the old code. This is a question for Andrew.